src/WopiValidator.Core/Mutators/ProofKeyMutator.cs

```diff
@@ -52,29 +52,34 @@ public string Name
 		}
 
 
-		public Dictionary<string, string> Mutate(Dictionary<string, string> originalProofKeyHeaders)
+		public Dictionary<string, string> Mutate(
+			Dictionary<string, string> originalProofKeyHeaders,
+			Func<long, Dictionary<string, string>> proofKeyGeneration)
```

Comment:
I'm not really excited about passing in a delegate like that, but it makes things much easier. Otherwise you'd have to pass everything that's needed to generate proofKey data (accessToken, uri and two RSACryptoServiceProviders) and have a way for the mutator to call into that logic, which current is not possible. Passing that as a delegate hides all that because all the required things can be captured automatically by the compiler via a closure.